common/models/change.js

```diff
@@ -186,27 +186,51 @@ module.exports = function(Change) {
 
     cb = cb || utils.createPromiseCallback();
 
-    change.currentRevision(function(err, rev) {
+    var model = this.getModelCtor();
+    var id = this.getModelId();
+
+    model.findById(id, function(err, inst) {
       if (err) return cb(err);
 
-      // avoid setting rev and prev to the same value
-      if (currentRev === rev) {
-        change.debug('rev and prev are equal (not updating anything)');
-        return cb(null, change);
-      }
+      change.tenant = getTenant(model, inst);
```

Comment:
First of all, I would like to acknowledge that I consider your use case, where you would like to copy certain model properties to each change record, as a valid one that we should support.

The question now is how to implement this feature in a way that's generic enough to be useful for other users of the framework too.

Can we perhaps build on top of my proposal in https://github.com/strongloop/loopback/pull/2959#discussion_r89064825, where I am describing a new method `PersistedModel.createChangeFilter` used when querying changes, and use a similar approach for the other part where we are creating/updating the change records? Something like `PersistedModel.createChangeData` that should supersede current [`Change.revisionForInstance`](https://github.com/strongloop/loopback/blob/f71030f8c2f61163b689f93e6086bfe1e4db9df6/common/models/change.js#L296-L299) method. The downside is that this proposal will require complex changes in several places, as we will have to stop passing a single property `revision` and start passing a set of properties instead.

Alternatively, can we perhaps keep your current design, but make it more flexible by calling a method on `PersistedModel` instead of calling a hard-coded `getTenant` method?

Here is a quick mock-up:

```js
// here in this method
model.findById(id, function(err, inst) {
  if (err) return cb(err);

  if (inst) {
    inst.fillChangeProperties(change);
  }

  change.currentRevision(inst, function(err, rev) { 
    // etc.
  });
});

// in lib/persisted-model.js

PersistedModel.prototype.fillChangeProperties = function(change) {
  // no-op by default
};
```

Can we come up with a better/more descriptive method name than `fillChangeProperties`?

Should this new method by async (accepting a callback arg) to allow the method to possibly fetch additional data?